### DISCUSSION TITLE
[JS SDK] Best Practice for determining if authWithOAuth2Code is a new user?

### DISCUSSION THREAD

**Discussion by jjgmckenzie** [pocketbase/pocketbase#2478] [Q&A] [answered]
Per https://github.com/pocketbase/pocketbase/discussions/1592 
>You'll have to send a second update() user request with the previously extracted OAuth2 fields you want to assign.

Given the same function is called whether a user is signing up or already has an account; is there a way to determine if this user already has "been initialized" so as to avoid repeating this process?

use case:

1. If they have an avatar on their gmail, you import it when they create an account to the application 
2. the user sets their avatar in the application's settings
3.  every time thereafter, when the user logs in, without a check whether or not they already have an account, they will reset their avatar to the initial, gmail one

$isNew does not seem to work for this purpose, as the record is already in the database when it is retrieved by Oauth

**Comment by ganigeorgiev** [comment, marked as answer]
There is a flag `IsCreate` but it is accessible only if you use the `OnRecordBeforeAuthWithOAuth2Request` or `OnRecordAfterAuthWithOAuth2Request` Go hooks. I'll consider exposing it in the json auth response with the next 0.16/0.17 release.

As a temporary workaround, you can detect whether the user has changed their settings already by comparing the `created` and `updated` fields.
